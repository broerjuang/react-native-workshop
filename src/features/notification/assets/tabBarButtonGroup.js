// @flow

import React, {Component} from 'react';
// import {View, Text} from 'react-native';
import {ButtonGroup} from 'react-native-elements';

type Props = {};

type State = {
  selectedIndex: number;
};
export default class tabBarButtonGroup extends Component<Props, State> {
  constructor() {
    super();
    this.state = {
      selectedIndex: 0,
    };
  }
  updateIndex(selectedIndex: number) {
    this.setState({selectedIndex});
  }

  render() {
    const buttons = ['Unread', 'Participating', 'All'];
    const {selectedIndex} = this.state;

    return (
      <ButtonGroup
        buttons={buttons}
        onPress={this.updateIndex.bind(this)}
        selectedIndex={selectedIndex}
      />
    );
  }
}
