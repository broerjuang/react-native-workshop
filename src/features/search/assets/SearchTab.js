// @flow

import React, {Component} from 'react';
import {View, TextInput, TouchableOpacity, StyleSheet} from 'react-native';
import {ButtonGroup} from 'react-native-elements';
import type {NavigationScreenProp} from 'react-navigation';
import {MaterialIcons, MaterialCommunityIcons} from '@expo/vector-icons';

type Props = {
  navigation: NavigationScreenProp<[]>;
};
type State = {
  selectedIndex: number;
  searchInput: string;
  showClear: boolean;
};

class SearchTab extends Component<Props, State> {
  constructor() {
    super();
    this.state = {
      selectedIndex: 0,
      searchInput: '',
      showClear: false,
    };
  }

  render() {
    const buttons = ['Repositories', 'Users'];
    const {selectedIndex, searchInput, showClear} = this.state;

    const {inputContainer, inputText} = styles;
    return (
      <View style={{backgroundColor: '#fff', marginTop: 20}}>
        <View style={inputContainer}>
          {searchInput === '' ? (
            <TouchableOpacity onPress={() => this.clearInput()}>
              <MaterialIcons name="search" size={20} color="grey" />
            </TouchableOpacity>
          ) : (
            <TouchableOpacity onPress={() => this.clearInput()}>
              <MaterialCommunityIcons
                name="arrow-left"
                size={20}
                color="grey"
              />
            </TouchableOpacity>
          )}
          <TextInput
            style={inputText}
            value={searchInput}
            onFocus={() => this.toggleClear()}
            onBlur={() => this.toggleClear()}
            onChangeText={(text) => this.doSearch(text)}
          />
          {showClear ? (
            <TouchableOpacity onPress={() => this.clearInput()}>
              <MaterialIcons name="clear" size={20} color="grey" />
            </TouchableOpacity>
          ) : null}
        </View>
        <ButtonGroup
          onPress={this.updateIndex}
          selectedIndex={selectedIndex}
          buttons={buttons}
        />
      </View>
    );
  }

  updateIndex = (selectedIndex: number) => {
    this.setState({selectedIndex}, () => {
      let screen = selectedIndex === 0 ? 'Repositories' : 'Users';
      this.props.navigation.navigate(screen);
    });
  };

  toggleClear = () => {
    this.setState({showClear: !this.state.showClear});
  };

  clearInput = () => {
    this.setState({searchInput: ''});
  };

  doSearch = (text: string) => {
    this.setState({searchInput: text});
  };
}

const styles = StyleSheet.create({
  inputContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    margin: 6,
    paddingVertical: 2,
    paddingHorizontal: 6,
    borderRadius: 5,
    backgroundColor: '#efefef',
  },
  inputText: {height: 40, width: '90%', backgroundColor: '#efefef'},
});

export default SearchTab;
