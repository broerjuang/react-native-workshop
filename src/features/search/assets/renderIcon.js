// @flow
import React, {Component} from 'react';
import {MaterialCommunityIcons} from '@expo/vector-icons';

type IconProps = {
  name: string;
  size: number;
  tintColor: string;
};

function renderIcon(props: IconProps) {
  let {name, size, tintColor} = props;
  return <MaterialCommunityIcons name={name} size={size} color={tintColor} />;
}

export default renderIcon;
