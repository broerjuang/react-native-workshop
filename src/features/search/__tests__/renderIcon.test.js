import renderer from 'react-test-renderer';
import React from 'react';
import {renderIcon} from '../assets/renderIcon';

describe('renderIcon test', () => {
  it('should render icon correctly', () => {
    let props = {
      name: 'home',
      size: 32,
      tintColor: '#000',
    };
    let component = renderer.create(<renderIcon {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
