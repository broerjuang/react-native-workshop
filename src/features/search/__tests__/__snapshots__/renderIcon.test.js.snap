// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderIcon test should render icon correctly 1`] = `
<renderIcon
  name="home"
  size={32}
  tintColor="#000"
/>
`;
